### DISCUSSION TITLE
Is there a way to perform different submit handlers based on a <button type="submit" name="..."> ?

### DISCUSSION THREAD

**Discussion by chvanam** [TanStack/form#886] [Q&A] [answered]
Or should I have a form field value for the submit button to be able to track it

**Comment by Balastrong** [comment]
It is currently not supported but it's something we can integrate to the library.

What is your usecase and what kind of API would you like to see?

**Comment by chvanam** [comment]
According to the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#name) for the `<button>` element, when a `<button type="submit">` has a `name` and optional `value` attribute and triggers a form submission,  it also add the `name/value` pair as part of the submitted form data.

In such case, i'm guessing that it could be handled with the existing TanStack Form API by adding an optional value for that button, but then we are just missing a way to execute `field.handleChange(e.target.value)`'s equivalent for a button's `onClick`, however there is no easy way to get that value (`e.target.value` doesn't exist for a `button[name=]` `onClick` event)

A higher level idea could be to add an optional `name` value on the top level `form` `onSubmit()` API, that is set if the form is submitted by a `button` with a `name` and optional `value`. But in that case i'm not sure on how to get the best type safety and inference which is a primary focus of this great library..

I'll keep thinking a bit but I'm open to back-and-forth API discussions here!

**Comment by chvanam** [comment]
Linked issue #890

**Comment by crutchcorn** [comment, marked as answer]
https://github.com/TanStack/form/issues/890#issuecomment-2312831802
